### ISSUE TITLE
[Bug/Feature]: Fix distributed='auto' override bug and complete DDP integration (follow-up to #592)

### ISSUE THREAD

**Issue by Adit-Jain-srm** [Adit-Jain-srm/NightmareNet#608]
## Problem

PR #592 wired the distributed training module into the Trainer but left several acceptance criteria from #371 unmet, plus introduced a bug in the CLI override logic.

### Bug: `distributed="auto"` forces distributed mode

`trainer.py` L319: `if distributed is not None: self.use_distributed = True` fires when `distributed="auto"`, overriding `training.distributed: false` in config. The intent of "auto" is device auto-detection, not forcing DDP on.

**Fix:**
```python
if distributed is not None and distributed != "auto":
    self.use_distributed = True
```

### Missing: Gradient accumulation world-size scaling

When running with N GPUs, effective batch size = per_device_batch * world_size. The trainer currently does not scale gradient accumulation steps or batch size based on `world_size`. This means per-GPU batch size is unchanged, silently changing the effective batch size.

### Missing: Rank-aware checkpointing

The existing `AtomicCheckpointer` saves on every rank. Under DDP, only rank 0 should write checkpoints (all ranks load). Without this, N processes write the same file concurrently, risking corruption.

### Missing: Real `accelerate launch` integration test

The current test (`test_trainer_integration.py`) uses full mocking. A real integration test should run `accelerate launch --num_processes=1` to verify single-process DDP works end-to-end without error.

## Acceptance Criteria

- [ ] Fix: `distributed="auto"` does NOT force `self.use_distributed = True` when config says `false`
- [ ] Gradient accumulation steps are divided by `world_size` (or effective batch documented)
- [ ] `AtomicCheckpointer.save()` only writes on rank 0; all ranks call `load()`
- [ ] Integration test uses `accelerate launch --num_processes=1` (not full mocking)
- [ ] All existing distributed tests continue to pass

## Scope

- `nightmarenet/training/trainer.py` (CLI override fix + batch scaling)
- `nightmarenet/training/checkpointer.py` (rank-aware save)
- `tests/test_trainer_integration.py` (real accelerate test)

## References

- Parent issue: #371
- PR #592 (partial implementation, now merged)

> **Important:** Please read CONTRIBUTING.md in full before requesting assignment. Assignment requests without a technical approach will not be considered. PRs demonstrating thorough research, benchmarks, or creative approaches beyond minimum criteria may earn bonus XP labels.

**Comment by Ayush-0918** [comment]
Please assign to me.

**Comment by Ruchira807** [comment]
@Adit-Jain-srm would like to work on this issue.

**Comment by Adit-Jain-srm** [comment]
Assigned @Ayush-0918
